### PR TITLE
Clean up itinerary summary rows

### PR DESCRIPTION
**Pull request by csolem**
Suggestion for implementation of the design for the itinerary summary rows. Comments appreciated.

### CODE REVIEW

**Comment by samuliheljo** [comment]
Thanks!
Some improvements that should be made before merging.
I add review comments shortly.

File: app/component/summary/summary-row.scss [hunk 1]
```diff
@@ -1,151 +1,125 @@
...
+    line-height: 2;
+    width: 4em;
+  }
 
-.summary-circle.ferry::before {
-  content: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' width='30' height='30'><circle stroke-width='3' stroke='#{rgba($ferry-color, 0.9999999)}' fill='white' cx='15' cy='15' r='3'/></svg>");
-}
+  & .itinerary-legs {
+    //position: relative;
+    width: 200px;
```
Comment (samuliheljo) [inline app/component/summary/summary-row.scss]: Itinerary leg should not be fixed to 200px and it should stretch

File: app/component/summary/summary-row.cjsx [hunk 2]
```diff
@@ -1,103 +1,64 @@
...
+      <div className="itinerary-start-time">
+        {startTime.format("HH:mm")}
+      </div>
       <div className="itinerary-legs">{legs}</div>
-      <div className="itinerary-leg-times">{legTimes}</div>
-      <br/>
+      <div className="itinerary-end-time">
+        {endTime.format("HH:mm")}
+      </div>
+      <div className="action-arrow">
```
Comment (samuliheljo) [inline app/component/summary/summary-row.cjsx]: This seems pretty tight on my iPhone (arrow very close to right edge), so it requires some padding.

File: app/component/summary/summary-row.cjsx [hunk 3]
```diff
@@ -1,103 +1,64 @@
...
       "itinerary-summary-row"
       "cursor-pointer"
       passive: @props.passive
     ]
 
-    <div className={cx classes} onClick={() => @props.onSelect(@props.hash)}>
+    <div className={classes} onClick={() => @props.onSelect(@props.hash)}>
+      <div className="itinerary-duration-and-distance">
+        <div className="itinerary-duration">
+          {moment.duration(duration).humanize()}
```
Comment (samuliheljo) [inline app/component/summary/summary-row.cjsx]: We should use here pure minutes as in here:
https://projects.invisionapp.com/share/5X5RNXGGU#/screens/136398565

File: app/component/summary/summary-row.cjsx [hunk 4]
```diff
@@ -1,103 +1,64 @@
...
+        </div>
+        <div className="itinerary-walking-distance">
+          <Icon img={'icon-icon_walk'} viewBox={"6 0 40 40"}/>
+          {geoUtils.displayDistance(data.walkDistance)}
+        </div>
+      </div>
+      <div className="itinerary-start-time">
+        {startTime.format("HH:mm")}
+      </div>
```
Comment (samuliheljo) [inline app/component/summary/summary-row.cjsx]: Somewhere around here we should show start time for the first leg departure as in here:
https://projects.invisionapp.com/share/5X5RNXGGU#/screens/136398565

If I remember correctly your wish was to make it configurable. You can implement it like that if you want.

**Comment by samuliheljo** [comment]
plain image
img

File: app/component/summary/summary-row.scss [hunk 5]
```diff
@@ -1,151 +1,125 @@
 .itinerary-summary-row {
-  padding-left: 0.5em;
```
Comment (samuliheljo) [inline app/component/summary/summary-row.scss]: Row's height should be 50px

File: app/component/summary/summary-row.scss [hunk 6]
```diff
@@ -1,151 +1,125 @@
...
-.summary-circle.end::after {
-  width:15px;
-  height:15px;
-  position: absolute;
-  bottom: -1px;
-  right: 5px;
-  content: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' width='30' height='30'><circle stroke-width='2' stroke='white' fill='#{rgba($to-color, 0.9999999)}' cx='15' cy='15' r='4'/></svg>");
-}
+      .icon {
+        width: 1.75em;
```
Comment (samuliheljo) [inline app/component/summary/summary-row.scss]: This should be 20px x 20px

**Comment by samuliheljo** [comment]
I added comments. You might want to check those and verify details according to image above.
This might be useful: https://chrome.google.com/webstore/detail/page-ruler/jlpkojjdgbllmedoapgfodplfhcbnbpn?hl=en

**Comment by csolem** [comment]
Thanks, Samuli. Will go through the comments. We found that it does not work well on Sony Xperia Z3 + with five itinerary legs. Any design for five or more itinerary transit legs, especially for smaller devices?

**Comment by samuliheljo** [comment]
For smaller devices, where legs do not fit, our idea was to fade rest of the line like in front page:
screen shot 2016-04-01 at 11 19 13

**Comment by samuliheljo** [comment]
Thanks, this looks good. I'll make small style changes and after that merge it to master. Cheers!

**Comment by csolem** [comment]
Thanks. Let me know if there is something more I should do.

**Pull request merged**